src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -388,6 +392,13 @@ StartRemoteTransactionPrepare(struct MultiConnection *connection)
 
 	Assign2PCIdentifier(connection);
 
+	/* log transactions to workers in pg_dist_transaction */
```

Comment:
> The current recovery logic expects every remote prepared transaction to be logged separately.

I was really just commenting on the FindWorkerNode() - we could quite easily store that in the connection.

> I am a bit worried about frequent COPY commands generating a ton of records,

Yea, me too.  We probably should suggest calling the cleanup at a quite regular basis.

> We could also consider regressing a bit on transaction recovery here until we have an implementation that uses a more compact representation.

Not sure what you mean by that?